larray/core/group.py

```diff
@@ -1055,6 +1055,74 @@ def opmethod(self, other):
     __ne__ = _binop('ne')
     __eq__ = _binop('eq')
 
+    # TODO : test also group.axis?
+    def equals(self, other):
+        """
+        Checks if self is equal to another group.
+        Two groups are equal if they have the same name and key.
```

Comment:
Not really true. As you implemented it, two groups can be equal with different keys (as in your example below mixing IGroup & LGroup) and can be different with the same (slice) key (if the axes differ)